internal/services/apimanagement/api_management_backend_resource.go

```diff
@@ -540,17 +540,17 @@ func flattenApiManagementBackendCredentials(input *backend.BackendCredentialsCon
 	if input.Certificate != nil {
 		result["certificate"] = *input.Certificate
 	}
-	result["header"] = flattenApiManagementBackendCredentialsObject(*input.Header)
-	result["query"] = flattenApiManagementBackendCredentialsObject(*input.Query)
+	result["header"] = flattenApiManagementBackendCredentialsObject(input.Header)
+	result["query"] = flattenApiManagementBackendCredentialsObject(input.Query)
 	return append(results, result)
 }
 
-func flattenApiManagementBackendCredentialsObject(input map[string][]string) map[string]interface{} {
+func flattenApiManagementBackendCredentialsObject(input *map[string][]string) map[string]interface{} {
 	results := make(map[string]interface{})
-	if input == nil {
+	if input == nil || *input == nil {
```

Comment:
`*input == nil` won't be valid, since it'd be an empty slice, so this can become:

```suggestion
	if input == nil {
```

Reply:
yes, that what I have ever thinked about too, but added it to align with the old logic. updated to remove it now.